cabal-testsuite/PackageTests/NewBuild/CmdRun/ScriptWithProjectBlock/cabal.test.hs

```diff
@@ -0,0 +1,6 @@
+import Test.Cabal.Prelude
+
+main = cabalTest $ do
+    -- script is called "s.hs" to avoid Windows long path issue in CI
+    res <- cabal' "v2-run" ["s.hs"]
+    assertOutputContains "Hello World" res
```

Comment:
does this test if `optimization: 2` is being effectively applied? if that is not the case i would try to use a project flag which could make the compilation or the execution fail if not set (not sure if check if optimiztion is applied is easy to do)

Reply:
oh yeah the `Build profile: -w ghc-<GHCVER> -O2` will only be shown with that flag